lstgen/generators/go.py

```diff
@@ -0,0 +1,268 @@
+                wr.writeln('this.{} = value'.format(var.name))
+
+        # create getters for output vars
+        for var in self.parser.output_vars:
+            wr.nl()
+            signature = 'func (this *{cls}) Get{cap}() {type}'.format(
+                cls=self.class_name,
+                cap=var.name.capitalize(),
+                type=self._convert_vartype(var.type)
+            )
+            with wr.indent(signature):
+                wr.writeln('return this.{}'.format(var.name))
+
+        self._write_method(self.parser.main_method)
+        for method in self.parser.methods:
+            self._write_method(method)
+        wr.nl()
+
+    def _convert_vartype(self, vartype):
+        return {
+            'BigDecimal.ZERO': 'decimal.NewFromInt(0)',
+            'BigDecimal[]': '[]'+self.bd_class,
+            'BigDecimal': ''+self.bd_class,
+            'int': 'int64',
+            'double': 'int64',
+        }[vartype]
+
+    def _conv_list(self, node):
+        res = super(GoGenerator, self)._conv_list(node)
+        return ['[]'+self.bd_class]+res
+
+    def _conv_attribute(self, node):
+        clsmethod = False
+        if node.attr == 'valueOf':
+            node.attr = 'NewFromInt'
+            clsmethod = True
+        elif node.attr == 'ZERO':
+            node.attr = 'NewFromInt(0)'
+            clsmethod = True
+        elif node.attr == 'ONE':
+            node.attr = 'NewFromInt(1)'
+            clsmethod = True
+        elif node.attr == 'TEN':
+            node.attr = 'NewFromInt(10)'
+            clsmethod = True
+        elif node.attr == 'longValue':
+            node.attr = 'IntPart'
+        elif node.attr == 'add':
+            node.attr = 'Add'
+        elif node.attr == 'subtract':
+            node.attr = 'Sub'
+        elif node.attr == 'multiply':
+            node.attr = 'Mul'
+        elif node.attr == 'divide':
+            node.attr = 'Div'
+        elif node.attr == 'compareTo':
+            node.attr = 'Cmp'
+        elif node.attr == 'setScale':
+            node.attr = 'Round'
+        elif node.attr in ('ROUND_UP', 'ROUND_DOWN'):
+            pass
+        else:
+            print("Warning: Unmapped attribute, probably broken result:", node.attr)
+        if clsmethod:
+            #print("NODE", node.value.id, node.attr)
+            return ['decimal', self.property_accessor_op, node.attr]
+        return (
+            self.to_code(node.value) +
+            [self.property_accessor_op, node.attr]
+        )
+
+    def _get_decimal_constructor_from_arg(self, arg):
+        try:
+            int(str(arg.n))
+            return 'NewFromInt'
+        except ValueError:
+            return 'NewFromFloat'
+        except AttributeError: # If node.args[0] is not a value, but e.g. a name or a call
+            # Unfortunately it's not possible to support all cases due to the strict
+            # typing of Go. However, we are "lucky":
+            if arg.__class__.__name__ == 'Name':
```

Comment:
Maybe combine it into `if arg.__class__.__name__ in ('Name', 'Call', 'BinOp'): ...`?